### PR TITLE
Marketplace tutorial improvements

### PR DESCRIPTION
**Pull request by joshuahannan**
Closes https://github.com/onflow/flow/issues/362

## Description

Updates the marketplace tutorial to use cadence best practices and the same contracts that are in the ft and nft tutorials

The updated code is in this PR: https://github.com/onflow/flow-playground-tutorials/pull/7

* Use `ExampleToken` and `ExampleNFT` instead of `FungibleToken` and `NonFungibleToken`
* Use correct storage paths
* Use more parameters in events
* Use a capability to the nft collection instead of storing nfts directly
* Add a lot more clarification about capabilities
* More links to nft storefront

______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Comment by joshuahannan** [comment]
@SupunS @dsainati1 Any idea why the CI isn't completing here?

**Comment by joshuahannan** [comment]
@turbolent Can you help me figure out why the lint and test steps aren't completing here? Thanks!

**Comment by turbolent** [comment]
@joshuahannan after you opened this PR, another PR changed the names of the CI jobs (e.g `test` -> `Test`). The project configuration expects the `Lint` and `Test` jobs to succeed, but this PR's CI config in this branch still has the outdated job names (which pass BTW). 

TLDR: Merge master back into your branch (`git merge master`) and push the changes up. That will retrigger CI and have everything pass. 

Sorry for the inconvenience, didn't notice before

**Review by turbolent — APPROVED** [review]
Didn't check the code changes in detail, mostly read the new guide text, which looks good 👍 \

File: docs/tutorial/06-marketplace-compose.mdx [hunk 1]
```diff
@@ -605,20 +797,22 @@ If we wanted to build a central marketplace on-chain, we could use a contract th
...
     // and returns the index of the sale so that users can know
     // how to remove it from the marketplace
-    pub fun listSaleCollection(collection: Capability<&SaleCollection>): Int {
-        self.tokensForSale.append(collection)
-        return (self.tokensForSale.length - 1)
+    pub fun listSaleCollection(collection: Capability<&SaleCollection>) {
+        let saleRef = collection.borrow()
+            ?? panic("Invalid sale collection capability")
+
+        self.tokensForSale[saleRef.owner?.address] = collection
```
Comment (turbolent) [inline docs/tutorial/06-marketplace-compose.mdx:809]: Does this type-check? Indexing using an optional into the dictionary shouldn't be possible. Maybe you meant
```suggestion
        self.tokensForSale[saleRef.owner!.address] = collection
```

Comment (joshuahannan) [reply docs/tutorial/06-marketplace-compose.mdx]: that code isn't meant to actually be used, just a little example, but that is a good point. I fixed the line. :)
